plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -112,8 +105,31 @@ func UTF16ToUTF8Bytes(in []byte, length uint32) ([]byte, error) {
 	return decoded, err
 }
 
+func UTF16ToUTF8BytesForWindowsEventBuffer(in []byte, length uint32) ([]byte, error) {
+	// Since Windows server 2022, the returned value of used buffer represents for double bytes char count,
+	// which is half of the actual buffer used by byte(what older Windows OS returns), checking if the length
+	//land on the end of used buffer, if no, double it.
+	if NumbersOfBytesPerCharacter == UnknownBytesPerCharacter {
+		if isTheEndOfContent(in, length) {
+			log.Printf("I! Buffer used: %d is returning as single byte character count", length)
+			NumbersOfBytesPerCharacter = 1
+		} else {
+			log.Printf("I! Buffer used: %d is returning as double byte character count, doubling it to get the whole buffer content.", length)
+			NumbersOfBytesPerCharacter = 2
+		}
+	}
+
+	i := int(length) * NumbersOfBytesPerCharacter
+
+	if i > cap(in) {
+		i = cap(in)
+	}
+
+	return UTF16ToUTF8Bytes(in, uint32(i))
+}
+
 func isTheEndOfContent(in []byte, length uint32) bool {
-	// scan next 20 bytes, if any of them is none '0', return false
+	// scan next 100 bytes, if any of them is none '0', return false
```

Comment:
Honestly I think rather than constantly updating this value in the comment, we could just refer to the variable name. Changes here are noisy and don't really add a lot of value - also it can get out of sync pretty easily